service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/HearingBookingDetailsController.java

```diff
@@ -103,7 +104,7 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
     @PostMapping("/submitted")
     public void handleSubmittedEvent(@RequestBody CallbackRequest callbackRequest) {
         CaseData caseData = mapper.convertValue(callbackRequest.getCaseDetails().getData(), CaseData.class);
-        if (standardDirectionsService.hasEmptyDates(caseData)) {
+        if (caseData.getState() == GATEKEEPING && standardDirectionsService.hasEmptyDates(caseData)) {
```

Comment:
Not sure if this field is updated automatically by CCD (would assume so but haven't tested it for anything other than changing the case state) but the `caseDetails.getState()` definitely is

Reply:
@joez-f does it mean we are changing that code or what we do~?

Reply:
@akomisarek turns out that CCD doesn't update that field so it is only used for updating the case state from a controller and nothing more. If you want to work out the current state you need to get the state from case details, which is annoying as that is as a `String` and not as our `State` enum.

Reply:
have changed it now, I thought caseData.getState() actually worked when I tested but lewis is right it does not work, I didn't test every scenario (doug spotted it though!)